### PR TITLE
[OPS-887] Switch to stack instead of cabal

### PR DESCRIPTION
**Pull request by zhenyavinogradov**
Haskell.nix is updated to use stack as well. When using stack directly, you have to use stack 1.x to make `lootbox` dependency work, it is available in `nix-shell tools.nix`.

I really don't like it, I think we should avoid using stack actually.

### CODE REVIEW

**Comment by zhenyavinogradov** [comment]
We should either go with this one, or stay with cabal and apply https://github.com/serokell/nixfmt-bot/pull/9 to update haskell.nix.

The problem with cabal is that the `lootbox` dependency does not have `.cabal` files in the repository and cabal does not support that very well, but @Haskell-mouse mentioned he might get rid of this dependency. And even if the dependency stays, I think we should find other ways to make it work without switching to stack.

File: nixfmt-bot.nix [hunk 1]
```diff
@@ -2,19 +2,19 @@ let
   buildDepError = pkg:
     builtins.throw ''
       The Haskell package set does not contain the package: ${pkg} (build dependency).
-
+
```
Comment (mkaito) [inline nixfmt-bot.nix:5]: Please configure your editor to delete trailing whitespace :wink:

Comment (zhenyavinogradov) [reply nixfmt-bot.nix:5]: That's an autogenerated file :)

File: tools.nix [hunk 2]
```diff
@@ -0,0 +1,15 @@
+let
```
Comment (mkaito) [inline tools.nix:1]: Why did you not call it `shell.nix`?

Comment (zhenyavinogradov) [reply tools.nix:1]: `shell.nix` is already used by haskell.nix to give you ghc with all dependencies of your project. Having this file separately gives you `stack` and `nix-tools` without having to build all the dependencies

**Pull request closed without merging**